store/postgres/src/copy.rs

```diff
@@ -744,6 +754,24 @@ impl Connection {
                 if table.is_cancelled(&self.conn)? {
                     return Ok(Status::Cancelled);
                 }
+
+                // Pause copying if replication is lagging behind to avoid
+                // overloading replicas
+                let mut lag = catalog::replication_lag(&self.conn)?;
+                if lag > MAX_REPLICATION_LAG {
+                    loop {
+                        info!(&self.logger,
+                             "Replicas are lagging too much; pausing copying for {}s to allow them to catch up",
+                             REPLICATION_SLEEP.as_secs();
+                             "lag_s" => lag.as_secs());
+                        std::thread::sleep(REPLICATION_SLEEP);
```

Comment:
Isn't sleeping for 30s excessive here before polling again? I'd imagine there's many cases where the replica catches up within that time. Anyway, this is something that we can measure and tweak later.

Reply:
Yeah, true; I am lowering this to 10s. Querying for lag every 10s shouldn't cause issues

Reply:
:+1: